docs/intro.asciidoc

```diff
@@ -6,20 +6,30 @@ endif::[]
 [[intro]]
 == Introduction
 
-Welcome to the APM .NET Agent documentation.
+The Elastic APM .NET Agent automatically measures the performance of your application and tracks errors.
+It has built-in support for the most popular frameworks,
+as well as a simple API which allows you to instrument any application.
 
-Elastic APM automatically measures the performance of your application and tracks errors.
-For example, it records spans for database queries and transactions for incoming HTTP requests.
-See <<supported-technologies>> to learn about which frameworks and technologies are supported by Elastic APM .NET agent.
+[float]
+[[how-it-works]]
+=== How does the Agent work?
 
-Spans are grouped in transactions - by default one for each incoming HTTP request.
-But it's possible to create custom transactions not associated with an HTTP request.
+The agent auto-instruments <<supported-technologies,supported technologies>> and records interesting events, like HTTP requests and database queries.
+To do this, it uses built-in capabilities of the instrumented frameworks like
+https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.diagnosticsource?view=netcore-3.0[Diagnostic Source],
+an HTTP module for IIS, or
+https://docs.microsoft.com/en-us/dotnet/api/system.data.entity.infrastructure.interception.idbcommandinterceptor?view=entity-framework-6.2.0[IDbCommandInterceptor] for Entity Framework.
+This means that for the supported technologies, there are no code changes required beyond enabling <<setup,auto-instrumentation>>.
 
-By default the agent comes with support for ASP.NET Core out-of-the-box.
-To instrument other events, you can use custom spans.
+The Agent automatically registers callback methods for built-in Diagnostic Source events.
+The IIS HTTP module, as described in <<setup-asp-net, Set up ASP.net Core>>, will need to be registered.
```

Comment:
One last thing: The IIS HTTP module is only required for ASP.NET Classic - it's not needed for ASP.NET Core. So what I tried to say here is that for almost everything the agent registers for the built-in Diagnostic Source events, except in case of ASP.NET Classic - in that case you need to manually register the IIS module, because there is no Diagnostic Source for that - so once either the the Diagnostic Source listeners of the HTTP module is registered, agent code will be triggered and measure the duration, etc. 

I'm not even sure we need to go into these details, I just wanted to mention the IIS module for completeness, but maybe it was rather confusing and less useful.


I let you decide, but I think it'd make sense to either drop the IIS module sentence, or adapt it as described above.

Reply:
Thanks for the clarification. Let's remove the sentence. It's probably more detail than is necessary.